docker/pypi/wmagent/bin/manage-common.sh

```diff
@@ -39,14 +39,71 @@ _exec_mysql() {
     else
         mysql -sN -u $MYSQL_USER --password=$MYSQL_PASS -h $MYSQL_HOST --execute="$sqlStr"
     fi
-}
 
+    ## TODO: To add the same functionality for recognizing the type of call, similar to _exec_oracle
+    #
+    # if $isPipe || $noArgs
+    # then
+    #     mysql -u $MYSQL_USER --password=$MYSQL_PASS -h $MYSQL_HOST --database=$wmaDBName --pager='less -SFX'
+    # else
+    #     local sqlStr=$1
+    #     local dbName=$2
+    #     if [[ -n $dbName ]]; then
+    #         mysql -sN -u $MYSQL_USER --password=$MYSQL_PASS -h $MYSQL_HOST --database=$dbName --execute="$sqlStr"
+    #     else
+    #         mysql -sN -u $MYSQL_USER --password=$MYSQL_PASS -h $MYSQL_HOST --execute="$sqlStr"
+    #     fi
+    # fi
+}
 
 _exec_oracle() {
     # Auxiliary function to avoid repetitive and long calls to the sqlplus command
-    echo "Not implemented"
-}
+    # :param: $@ could be a sql string to execute or a file redirect or a here document
+
+    # We check for input arguments
+    local execStr=""
+    if [[ -z $* ]]
+    then
+        local hasArgs=false
+    else
+        local hasArgs=true
+        # Building a default executable string:
+        execStr="$execStr SET HEADING OFF;\n"
+        execStr="$execStr SET ECHO OFF;\n"
+        execStr="$execStr SET UNDERLINE OFF;\n"
+        execStr="$execStr SET LINESIZE 1024;\n"
+        execStr="$execStr SET FEEDBACK OFF;\n"
+        execStr="$execStr SET PAGESIZE 0;\n"
+        execStr="$execStr whenever sqlerror exit sql.sqlcode;\n"
+        execStr="$execStr $@"
+        execStr="${execStr%;};\n"
+        execStr="$execStr exit;\n"
+    fi
 
+    # First we need to know if we are running through a redirected input
+    # if fd 0 (stdin) is open and refers to a terminal - then we are running the script directly, without a pipe
+    # if fd 0 (stdin) is open but does not refer to the terminal - then we are running the script through a pipe
+    # NOTE: Docker by default redirects stdin
+    local isPipe=false
+    if [[ -t 0 ]] ; then isPipe=false; else isPipe=true ; fi
+
+    # Then we traverse the callstack to find if the original caller was init.sh
+    # if so - we never redirect
+    local isInitCall=false
+    for callSource in ${BASH_SOURCE[@]}
```

Comment:
I am not sure I understand this. How is this `callstack` supposed to look like? What is and what does it mean init.sh to be a "caller"?